fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -5349,6 +5549,84 @@ bool DDTeamCollection::notEnoughTeamsForAServer() const {
 	return false;
 }
 
+std::vector<size_t> DDTeamCollection::getHealthyStorageServerCountPerMachine() const {
+	std::unordered_map<std::string, int> serverCountOnMachines;
+	size_t serverCount = 0;
+	for (auto& [serverID, server] : server_info) {
+		if (server_status.get(serverID).isUnhealthy()) {
+			continue;
+		}
+		std::string machineID = server->machine->machineID.toString();
+		if (serverCountOnMachines.find(machineID) == serverCountOnMachines.end()) {
+			serverCountOnMachines[machineID] = 0;
+		}
+		serverCountOnMachines[machineID] = serverCountOnMachines[machineID] + 1;
+		serverCount++;
+	}
+	size_t maxCount = 0;
+	size_t minCount = std::numeric_limits<size_t>::max();
+	for (const auto& [id, count] : serverCountOnMachines) {
+		if (count > maxCount) {
+			maxCount = count;
+		}
+		if (count < minCount) {
+			minCount = count;
+		}
+	}
+	return { minCount, maxCount, serverCountOnMachines.size(), serverCount };
+}
+
+std::vector<size_t> DDTeamCollection::getHealthyMachineCountPerZoneOrDataHall() const {
+	std::unordered_map<std::string, std::unordered_set<std::string>> machinesPerZone;
+	std::unordered_set<std::string> machines;
+	for (auto& [serverID, server] : server_info) {
+		if (!isMachineHealthy(server->machine)) {
+			continue;
+		}
+		std::string zoneId;
+		if (configuration.storagePolicy->info().find("data_hall") != std::string::npos) {
+			ASSERT(server->getLastKnownInterface().locality.dataHallId().present());
+			zoneId = server->getLastKnownInterface().locality.dataHallId().get().toString();
+		} else if (server->getLastKnownInterface().locality.zoneId().present()) {
+			zoneId = server->getLastKnownInterface().locality.zoneId().get().toString();
+		} else {
+			zoneId = server->machine->machineID.toString();
+		}
+		std::string machineID = server->machine->machineID.toString();
+		machinesPerZone[zoneId].insert(machineID);
+		machines.insert(machineID);
+	}
+	size_t maxCount = 0;
+	size_t minCount = std::numeric_limits<int>::max();
+	for (const auto& [id, machines] : machinesPerZone) {
+		size_t machineCount = machines.size();
+		if (machineCount > maxCount) {
+			maxCount = machineCount;
+		}
+		if (machineCount < minCount) {
+			minCount = machineCount;
+		}
+	}
+	return { minCount, maxCount, machinesPerZone.size(), machines.size() };
+}
+
+std::vector<std::string> DDTeamCollection::getServersOnMachineTeamDescription(
+    const std::vector<Reference<TCMachineInfo>>& machines) const {
+	std::vector<std::string> res;
+	for (auto& machine : machines) {
+		int healthySSCount = 0;
+		for (auto it : machine->serversOnMachine) {
+			if (!server_status.get(it->getId()).isUnhealthy()) {
```

Comment:
I think there is `isHealthy()` that can be used.

Reply:
Currently, we do not have such method in TCServer. I feel it may not be worth to add this new method for this case.